File: src/main/java/org/hisp/dhis/rules/models/RuleEffects.java
package org.hisp.dhis.rules.models;

import javax.annotation.Nonnull;
import java.util.List;

public record RuleEffects(
    @Nonnull TrackerObjectType trackerObjectType,
    @Nonnull String trackerObjectUid,
    @Nonnull List<RuleEffect> ruleEffects)
{
    public boolean isEnrollment(){
        return this.trackerObjectType == TrackerObjectType.ENROLLMENT;
    }

    public boolean isEvent(){
        return this.trackerObjectType == TrackerObjectType.EVENT;
    }
}

File: src/jvmMain/kotlin/org/hisp/dhis/rules/models/RuleEffects.kt
package org.hisp.dhis.rules.models
data class RuleEffects(
    val trackerObjectType: TrackerObjectType,
    val trackerObjectUid: String,
    val ruleEffects: List<RuleEffect>
) {
    val isEnrollment: Boolean
        get() = trackerObjectType == TrackerObjectType.ENROLLMENT
    val isEvent: Boolean
        get() = trackerObjectType == TrackerObjectType.EVENT

    fun ruleEffects(): List<RuleEffect> {
        return ruleEffects
    }

    fun trackerObjectUid(): String {
        return trackerObjectUid
    }
}
